### PR TITLE
Fix for CVE-2023-39325

### PR DESCRIPTION
**Pull request by lsm5**
Updated golang.org/x/net to v0.17.0

Ref: https://github.com/advisories/GHSA-4374-p667-p6c8
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2248243

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Fix for CVE-2023-39325

#### How to verify it

Check that golang.org/x/net/http2 is bumped to v0.17.0

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
Updated golang.org/x/net/http2 to v0.17.0. Security fix for CVE-2023-39325
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -141,3 +141,5 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	k8s.io/klog v1.0.0 // indirect
 )
+
+replace golang.org/x/net v0.15.0 => golang.org/x/net v0.17.0
```
Comment (rhatdan) [inline go.mod:145]: Why are you doing a replace here?

Comment (lsm5) [reply go.mod:145]: x/net is listed as an indirect dependency, that's why I did the replace. Would it be ok to directly update an indirect dep?

Comment (lsm5) [reply go.mod]: fixed.

**Comment by lsm5** [comment]
@TomSweeneyRedHat PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons

**Pull request merged**